### PR TITLE
feat(android): material you themes

### PR DESCRIPTION
**Pull request by m1ga**
Creating 3 new themes:
* Theme.Titanium.Material3.DayNight
* Theme.Titanium.Material3.DayNight.NoTitleBar
* Theme.Titanium.Material3.DayNight.Fullscreen
in order to use Material3 Dynamic Color themes. 

Adding `Ti.UI.Android.harmonizedColor()` to find a "harmonizing color"

```js
const win1 = Ti.UI.createWindow({});
const win2 = Ti.UI.createWindow({});

win1.add(Ti.UI.createButton({title: 'I am button.'}));
win2.add(Ti.UI.createButton({title: 'I am button.'}));

var tab1 = Ti.UI.createTab({
    window: win1,
    title: 'Blue',
		icon: "/appicon.png"
}),
tab2 = Ti.UI.createTab({
    window: win2,
    title: 'Red',
		icon: "/appicon.png"
}),
tabGroup = Ti.UI.createTabGroup({
    tabs: [tab1, tab2],
		shiftMode: false,
		style:  Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION,
		theme:"Theme.Titanium.Material3.DayNight.NoTitleBar"
});
tabGroup.open();
```

Screenshot_20221008-195624

```js
const win1 = Ti.UI.createWindow({});
var col1 = "#044";
var col2 = Ti.UI.Android.harmonizedColor(col1);
var col3 = Ti.UI.Android.harmonizedColor(col2);
var v1 = Ti.UI.createView({backgroundColor: col1, width: 100, height: 100, left: 0})
var v2 = Ti.UI.createView({backgroundColor: col2, width: 100, height: 100, left: 100})
var v3 = Ti.UI.createView({backgroundColor: col3, width: 100, height: 100, left: 200})
win1.add([v1,v2,v3]);
win1.open();
```

Screenshot_20221008-201949

**Alternative**
you can also create an own theme with `<style name="AppTheme" parent="Theme.Material3.DynamicColors.DayNight" />`

**Info**
The other PR #13028 maps dynamic colors to existing Ti colors with the OLD Material themes. Since there are more color slots now (see https://material.io/blog/migrating-material-3 `Colors and Themes`) it won't use e.g. `colorPrimaryContainer` with the old PR resulting in some different colors. I think if you want to use dynamic colors you want to use Material 3 themes too. So with this PR it will do that.

**Todo**
Add it to the titanium-docs Android theme list.

### CODE REVIEW

File: android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java [hunk 1]
```diff
@@ -867,6 +869,14 @@ private static void ensureExternalPublicMediaDirectoryExists()
 		}
 	}
 
+	@Kroll.method
+	public String harmonizerColor(String value)
```
Comment (hansemannn) [inline android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java:873]: I would rather see this under `Ti.UI` or `Ti.UI.Android`, what do you think? Also, I would prefer `harmonizedColor`

Comment (m1ga) [reply android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java:873]: both valid points :+1: will move it

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
Just to be sure: [This](https://github.com/tidev/titanium_mobile/commit/0c99bfe613dc161b95e853db228e7d811a80647a) were the changes for the last major theme changes (for dark mode support), maybe it includes something else useful. But since this one will not be the default new apps (yet), I think we're good.

**Comment by m1ga** [comment]
Exactly. If we move this to a main theme we have to (perhaps) do some changes. But since its a Material default theme now it might only be moving the main Titanium theme to a different parent.

**Pull request merged**